File: DFTmatrix.m
clear all, close all, clc
n = 1024;
w = exp(-i*2*pi/n);

% The slowlier way of computing the DFT matrix
% for i=1:n
%     for j=1:n
%         DFT[i,j]= w^((i-1)*(j-1));
%     end
% end

% The faster way of solving it 

[I,J] = meshgrid(1:n, 1:n);
DFT = w.^((I-1).*(J-1));
imagesc(real(DFT));

File: DFTmatrix.py
import numpy as np
import matplotlib.pyplot as plt

# this code equal the faster version in the DFT matrix matlab file
n = 1024
w = np.exp(-1j*2*np.pi/n)

J, K = np.meshgrid(np.arange(n), np.arange(n))
DFT = np.power(w, J*K)
DFT = np.real(DFT)

plt.imshow(DFT)
plt.show()
